lib/prism/ripper_compat.rb

```diff
@@ -299,6 +309,59 @@ def visit_rational_node(node)
       visit_number(node) { |text| on_rational(text) }
     end
 
+    # Visit a StringNode node.
+    def visit_string_node(node)
+      bounds(node.content_loc)
+      tstring_val = on_tstring_content(node.unescaped.to_s)
+      on_string_literal(on_string_add(on_string_content, tstring_val))
+    end
+
+    # Visit an XStringNode node.
+    def visit_x_string_node(node)
+      bounds(node.content_loc)
+      tstring_val = on_tstring_content(node.unescaped.to_s)
+      on_xstring_literal(on_xstring_add(on_xstring_new, tstring_val))
+    end
+
+    # Visit an InterpolatedStringNode node.
+    def visit_interpolated_string_node(node)
+      parts = node.parts.map do |part|
+        case part
+        when StringNode
+          bounds(part.content_loc)
+          on_tstring_content(part.content)
+        when EmbeddedStatementsNode
+          on_string_embexpr(visit(part))
+        else
+          raise NotImplementedError, "Unexpected node type in InterpolatedStringNode"
+        end
+      end
+
+      string_list = parts.inject(on_string_content) do |items, item|
+        on_string_add(items, item)
+      end
+
+      on_string_literal(string_list)
+    end
+
+    # Visit an EmbeddedStatementsNode node.
+    def visit_embedded_statements_node(node)
+      visit(node.statements)
+    end
+
+    # Visit a SymbolNode node.
+    def visit_symbol_node(node)
+      if node.opening && ['"', "'", "("].include?(node.opening[-1])
```

Comment:
I think we'll want to change this because there are a lot of possible bounds for `%s`.

```suggestion
      if (opening = node.opening) && (['"', "'"].include?(opening[-1]) || opening.start_with?("%s"))
```